packages/backend-common/src/reading/GitlabUrlReader.ts

```diff
@@ -188,8 +188,7 @@ export class GitlabUrlReader implements UrlReader {
       throw new Error(message);
     }
 
-    const commitSha = (await commitsGitlabResponse.json())[0].id;
-
+    const commitSha = (await commitsGitlabResponse.json())[0]?.id ?? '';
```

Comment:
Bit strange to have `{ etag: '' }` trigger a hit, perhaps keep this undefined and check for presence in the condition below?

Reply:
It's actually not so much about the condition; it's the fact that `treeResponseFactory.fromTarArchive` farther down has a non-optional `etag` argument. So in the current shape of things we have to pass in _something_ to it. Not sure that any other placeholder would make more sense than this either 🤔

Reply:
Alright, let's :shipit:. Won't be hitting the check below anyway, just looks scary x)